### PR TITLE
Add Deprecate components support

### PR DESCRIPTION
**Pull request by Glowbal**
**When merged this pull request will:**
- Add support for deprecating components
- Deprecrates ACE Sitting. Moved to ACEX Sitting.

### CODE REVIEW

File: addons/sitting/XEH_clientInit.sqf [hunk 1]
```diff
@@ -3,11 +3,12 @@
...
 
+GVAR(isEnabled) = false;
 ["SettingsInitialized", {
-    TRACE_1("SettingInit", GVAR(enable));
-
+    if ([[QUOTE(ADDON), QGVAR(enable)], ["acex_sitting", "acex_sitting_enable"], "3.8.0"] call EFUNC(common,deprecateComponent)) exitwith {};
     //If not enabled, then do not add CanInteractWith Condition or event handlers:
-    if (!GVAR(enable)) exitWith {};
+    if (!GVAR(enable) || GVAR(isEnabled)) exitWith {};
+    GVAR(isEnabled) = true;
```
Comment (jonpas) [inline addons/sitting/XEH_clientInit.sqf]: What is that for again?

Comment (Glowbal) [inline addons/sitting/XEH_clientInit.sqf]: Ensures that the module will not load in twice

Comment (jonpas) [inline addons/sitting/XEH_clientInit.sqf]: How could it? Unless someone put 2 of those down (in which case he should get all the RPT warnings).

Comment (Glowbal) [inline addons/sitting/XEH_clientInit.sqf]: You are quite right - this isn't necessary here, but it is for the port to ACEX. It should not have been commited with this PR.

Comment (jonpas) [inline addons/sitting/XEH_clientInit.sqf]: How are we going to handle that for components which have modules that can be placed multiple times (eg. slideshow)?

Comment (Glowbal) [inline addons/sitting/XEH_clientInit.sqf]: Same way we handle functions that we deprecate.

Comment (jonpas) [inline addons/sitting/XEH_clientInit.sqf]: Never mind, slideshow doesn't even have a XEH postInit, was thinking of something else.

**Comment by jonpas** [comment]
@Glowbal Will this print warnings for sitting? If so it has to be taken out before release.

**Pull request merged**